Modules/_vxwapi.c

```diff
@@ -0,0 +1,520 @@
+            _Py_write_noraise(errpipe_write, "noexec", 6);
+        }
+        /* We can't call strerror(saved_errno).  It is not async signal safe.
+         * The parent process will look the error message up. */
+    } else {
+        _Py_write_noraise(errpipe_write, "SubprocessError:0:", 18);
+        _Py_write_noraise(errpipe_write, err_msg, strlen(err_msg));
+    }
+
+    return Py_BuildValue("i", pid);
+}
+
+
+static void _save_fds(int saved_fd[])
+{
+    int fd;
+    int flags;
+
+    if (!saved_fd) return;
+
+    for (fd = 0; fd < FD_SETSIZE; ++fd) {
+        flags = fcntl (fd, F_GETFD);
+        if (flags < 0)
+            continue;
+
+        saved_fd[fd] = !(flags & FD_CLOEXEC);
+    }
+}
+
+static void _restore_fds(int saved_fd[])
+{
+    int fd;
+    int flags;
+
+    if (!saved_fd) return;
+
+    for (fd = 0; fd < FD_SETSIZE; ++fd) {
+        flags = fcntl (fd, F_GETFD);
+        if (flags < 0)
+            continue;
+
+        if (-1 != saved_fd[fd]) {
+            _Py_set_inheritable_async_safe(fd, saved_fd[fd], NULL);
+        }
+    }
+}
+
+/*[clinic input]
+_vxwapi.rtp_spawn
+
+    process_args: object
+    executable_list: object
+    close_fds: int
+    py_fds_to_keep: object(subclass_of='&PyTuple_Type')
+    cwd_obj: object
+    env_list: object
+    p2cread: int
+    p2cwrite: int
+    c2pread: int
+    c2pwrite: int
+    errread: int
+    errwrite: int
+    errpipe_read: int
+    errpipe_write: int
+    restore_signals: int
+    call_setsid: int
+    preexec_fn: object
+    /
+
+Spawn a real time process in the vxWorks OS
+[clinic start generated code]*/
+
+static PyObject *
+_vxwapi_rtp_spawn_impl(PyObject *module, PyObject *process_args,
+                       PyObject *executable_list, int close_fds,
+                       PyObject *py_fds_to_keep, PyObject *cwd_obj,
+                       PyObject *env_list, int p2cread, int p2cwrite,
+                       int c2pread, int c2pwrite, int errread, int errwrite,
+                       int errpipe_read, int errpipe_write,
+                       int restore_signals, int call_setsid,
+                       PyObject *preexec_fn)
+/*[clinic end generated code: output=5f98889b783df975 input=30419f3fea045213]*/
+{
+    PyObject *converted_args = NULL, *fast_args = NULL;
+    char *const *argv = NULL, *const *envp = NULL;
+    int saved_fd[FD_SETSIZE] = {-1};
+
+    PyObject *return_value = NULL;
+    if (_PyInterpreterState_Get() != PyInterpreterState_Main()) {
+        PyErr_SetString(PyExc_RuntimeError, "fork not supported for subinterpreters");
+        return NULL;
+    }
+
+    if (close_fds && errpipe_write < 3) {  /* precondition */
+        PyErr_SetString(PyExc_ValueError, "errpipe_write must be >= 3");
+        return NULL;
+    }
+
+    if (_sanity_check_python_fd_sequence(py_fds_to_keep)) {
+        PyErr_SetString(PyExc_ValueError, "bad value(s) in fds_to_keep");
+        return NULL;
+    }
+
+    if (preexec_fn != Py_None) {
```

Comment:
is there a benefit to attempting to match the `_posixsubprocess.fork_exec()` function signature?  It has a negative side effect on maintenance for everyone: When we want to add parameters to that, this code on a platform virtually nobody has access to would also need to be updated.

I'd like to keep the need to update the platform specific implementations isolated.